### DISCUSSION TITLE
why  when created System.Array is full of None in place of 0 ?

### DISCUSSION THREAD

**Discussion by echidne** [pythonnet/pythonnet#2381] [Q&A] [answered]
Hi,
I'm a beginner with pythonnet. 
when i create an array as : 
```
from System import Array
array = Array.CreateInstance(int,5)
```
that array is full of None

But if I do the equivalent operation in C# 
```
Array array=Array.CreateInstance( typeof(int), 5 );
```
or 
```
int[] array = new int[5];
```

the array is full of 0

Does my installation of pythonnet have a problem?
I found old tutorial of pythonnet on youtube and when they created array their array were full of zero and not None . So i'm wondering something is not working properly on my system

system: Windows 10
Python : Python 3.11 installed in a venv

**Comment by filmor** [comment, marked as answer]
In the first invocation, `int` is the Python `int` type, which is the object type `PyInt` in C#, so its `default` is `null` (`None` in Python). In C#, `int` is an alias for `System.Int32`, a value type with `default` 0.

**Reply by echidne** [reply]
Thank you.
I was following a tutorial from 2019  ([github](https://github.com/areed1192/sigma_coding_youtube/blob/master/python/python-pythonnet/Pythonnet%202%20-%20Fundamentals.ipynb), [video](https://youtu.be/J7TETPbLw7c?list=PLcFcktZ0wnNnz07eWc7N5ao1dyiXoV-ib&t=2186)) and visibly at that time PyInt default was 0.
I reproduce the actual C# behavior as : 
```
from System import Array, Int32

array_1 = Array.CreateInstance(Int32,5)
```
